### PR TITLE
Add governance treasury withdrawal test case

### PR DESCRIPTION
**Pull request by carbolymer**
# Description

Adds treasury withdrawal test case.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cardano-testnet/src/Testnet/Components/Query.hs [hunk 1]
```diff
@@ -77,26 +75,27 @@ waitUntilEpoch nodeConfigFile socketPath desiredEpoch = withFrozenCallStack $ do
...
+  :: MonadTest m
+  => MonadCatch m
+  => MonadIO m
+  => ExecConfig
+  -> NodeConfigFile In
+  -> SocketPath
+  -> EpochInterval  -- ^ Number of epochs to wait
+  -> m EpochNo -- ^ The epoch number reached
+waitForEpochs execConfig nodeConfigFile socketPath interval = withFrozenCallStack $ do
+  currentEpoch <- H.nothingFailM $ mEpoch <$> queryTip execConfig
```
Comment (Jimbo4350) [inline cardano-testnet/src/Testnet/Components/Query.hs:89]: So I don't think `NewEpochState` has the current `EpochNo` but it may be worth asking Alexy.

Comment (carbolymer) [reply cardano-testnet/src/Testnet/Components/Query.hs:97]: Is this comment still relevant?

**Review by Jimbo4350 — APPROVED** [review]
LGTM! A couple comments. In future, try to put changes not relevant to the PR description in a separate commit.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/TreasuryWithdrawal.hs [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    ]
+
+  void $ H.execCli' execConfig
+    [ eraName, "transaction", "submit"
+    , "--tx-file", stakeCertTxSignedFp
+    ]
+  -- }}}
+
+  -- {{{ Create treasury withdrawal
+  let withdrawalAmount = 3_300_777 :: Integer
```
[the remark this answers is not in the document]
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/TreasuryWithdrawal.hs:156]: No reason, I've used it in early stages when debugging, trying to spot in utxos.

**Review by smelc — APPROVED** [review]

File: cardano-testnet/src/Testnet/Components/Query.hs (hunk 1 above)
Comment (carbolymer) [reply cardano-testnet/src/Testnet/Components/Query.hs:99]: Will do in a follow-up PR.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/ProposeNewConstitution.hs [hunk 3]
```diff
@@ -256,25 +255,14 @@ hprop_ledger_events_propose_new_constitution = H.integrationWorkspace "propose-n
...
-    , "--volatile-tip"
-    , "--out-file", finalGovState
-    ]
-
   -- Tally registered votes
-
-  finalGovFileBS <- evalIO $ LBS.readFile finalGovState
-
-  decodedFinalGovFile <- H.nothingFail (Aeson.decode finalGovFileBS :: Maybe Aeson.Value)
+  govState <- getGovState execConfig ceo
```
[the remark this answers is not in the document]
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/ProposeNewConstitution.hs:259]: Will do in a follow-up PR.

**Pull request merged**